irohad/ordering/impl/ordering_service_impl.hpp

```diff
@@ -123,6 +124,16 @@ namespace iroha {
        */
       size_t proposal_height;
 
+      /**
+       * Mutex for proper quit handling
+       */
+      std::mutex m;
+
+      /**
+       * Set after destruction
+       */
+      bool is_finished;
```

Comment:
Maybe it is better to set atomic variable, if it is shared between the threads

Reply:
Doesn't help much